packages/back-end/src/models/ExperimentModel.ts

```diff
@@ -690,19 +709,35 @@ const logExperimentUpdated = async ({
   user,
   current,
   previous,
+  readAccessFilter,
 }: {
   organization: OrganizationInterface;
   user: EventAuditUser;
   current: ExperimentInterface;
   previous: ExperimentInterface;
+  readAccessFilter: ReadAccessFilter;
```

Comment:
Hmmm.... you are correct in your `Note` that if we don't pass the `project` object into `previousApiExperimentPromise` we could run into a situation where the project's `statsEngine` is different form the organizations `DEFAULT_STATS_ENGINE`.

For me, I'd prefer that our logs were 100% accurate, even if it means having to pass in an extra prop. I'd hate for an org to be using the logs to do some debugging and we introduce a red herring in the form of an incorrect `statsEngine`.

One other alternative, which I'm not 100% on, as it could be dangerous, is to create a new DB method on the `ProjectModel`. Something like `findProjectByIdDangerously`, where we don't take the user's permissions into account. 

Something like:
```ts
// only use this to retrieve the project object when you don't need to worry about the user's permission level
export async function findProjectByIdDangerously(
  id: string,
  organization: string,
) {
```

If we were to do something like that, we could reduce the number of places we have to pass down the `readAccessFilter`.

Reply:
> For me, I'd prefer that our logs were 100% accurate, even if it means having to pass in an extra prop. I'd hate for an org to be using the logs to do some debugging and we introduce a red herring in the form of an incorrect `statsEngine`.

I agree.

I think your suggestion for a 'dangerous' loader makes sense here. But I share your uncertainty around safeness. If we could guarantee that we would never log an experiment that belongs to a project that the user does not have access to, then it would seem fine. This PR isn't updating our Experiment models just yet, so it's hard to make a determination.

I'll keep thinking about this but unless you feel confident with the 'dangerous' fetching method it seems the move is to leave it as is.